CGO8: Base Google Maps Exploration

neliop opened the pull request:
Issue #8 complete:

- Enable location on GoogleMap widget
- Add My Location button that moves map to the user's location
- Switch Campus button now moves map to the campus that is furthest from the map's current camera position
- Remove previously existing bloc implementation
- Create Map bloc following the bloc package implementation
- Rename dart files to follow proper naming conventions

neliop committed:
Set up bloc packages
Set up bloc and flutter_bloc packages for easier and more structured bloc implementation
Enable location on map and add My Location button
Edit polygon style
Change all dart file names to proper naming convention

neliop committed:
Fix widget test and formatting

DBatistaFerreira commented on lib/widgets/component/quick_menu.dart:
```diff
@@ -37,54 +38,54 @@ class DrawerComponentState extends State<DrawerComponent> {
-                  .locationCoordinates
-                  .add(LatLng(concordia_constants.H_BUILDING_LATITUDE, concordia_constants.H_BUILDING_LONGITUDE));
+              bloc.add(CameraMove(
+                  LatLng(concordia_constants.H_BUILDING_LATITUDE, concordia_constants.H_BUILDING_LONGITUDE),
```
17.5 shows up 3+ times. Needs to be in a constant

neliop replied:
Fixed!

DBatistaFerreira commented on lib/blocs/map_bloc.dart:
```diff
@@ -0,0 +1,24 @@
+  ) async* {
+    if (event is CameraMove) {
+      if (event.setMarker == true) {
+        Set<Marker> markers = {Marker(markerId: MarkerId('some_marker'), position: event.coordinates)};
```
Some marker?

neliop replied:
Changed!

DBatistaFerreira commented on lib/utilities/concordia_constants.dart:
```diff
@@ -34,7 +34,10 @@ const String FG_BUILDING_INITIAL = "FG";
 const double FG_BUILDING_LONGITUDE = -73.578428;
 
-var buildingCoords = [
+const LatLng sgwCoordinates = LatLng(45.495835, -73.578789);
```
Do we wanna rename this SGW_COORDINATES?

DBatistaFerreira commented on lib/widgets/component/quick_menu.dart:
```diff
@@ -1,19 +1,20 @@
 import 'package:flutter/cupertino.dart';
```
I think were gonna have to refactor this class when we implement the menu for all buildings. We need a loop or something. Won't have to worry about it for now

MoreWaqar commented on lib/utilities/concordia_constants.dart:
```diff
@@ -34,7 +34,10 @@ const String FG_BUILDING_INITIAL = "FG";
+const LatLng sgwCoordinates = LatLng(45.495835, -73.578789);
+const LatLng loyolaCoordinates = LatLng(45.458257, -73.640450);
+
+var buildingCoordinates = [
```
May be worthwhile to combine this variable with the map below it to avoid having redundant data. Probably cleaner.

neliop replied:
Constants reorganized!

neliop committed:
Restructure building constants
Use LatLng to store building coordinates in the constants library
Add constants for zoom levels

DBatistaFerreira reviewed (approved):
Awesome work Nick! You're honestly doing great work!

MoreWaqar committed:
Fixed removal that broke the polygon onTap() method.

MoreWaqar reviewed (approved):
Great updates.
